linux/cuda_info.bsh

```diff
@@ -341,277 +353,335 @@ function cuda_capabilities()
-      CUDA_LTOS=("${CUDA_ARCHES[@]}")
       CUDA_DEPRECATED=(35 37 50)
-
-      if [ "${OS-}" == "Windows_NT" ]; then
-        CUDA_MINIMAL_DRIVER_VERSION=452.39
-      else
-        CUDA_MINIMAL_DRIVER_VERSION=450.80.02
-      fi
+      ;;
+    11.[5-7]*)
+      CUDA_ARCHES=(35 37 50 52 53 60 61 62 70 72 75 80 86 87)
+      CUDA_DEPRECATED=(35 37 50)
+      ;;
+    11.8*)
+      CUDA_ARCHES=(35 37 50 52 53 60 61 62 70 72 75 80 86 87 89 90)
+      CUDA_DEPRECATED=(35 37 50)
       ;;
 
-    # 11.7*) ?
+    # 11.9*) ?
     #   # run "strings `which nvcc` | grep compute_" (ARCHES) and "sm_" (CODES) and you'll get a complete list
     #   # The strings output typically has a "are deprecated and may be removed in a future release" message
     #   #
     #   # strings `command -v nvcc` | \grep '^sm_' | sed 's|sm_||' | tr '\n' ' '; echo
     #   # strings `command -v nvcc` | \grep '^compute_' | sed 's|compute_||' | tr '\n' ' '; echo
+    #   # strings `command -v nvcc` | \grep '^lto_' | sed 's|compute_||' | tr '\n' ' '; echo
     #
     #   CUDA_ARCHES=(?)
-    #   CUDA_CODES=("${CUDA_ARCHES}")
+    #   CUDA_CODES=(?) # only if different, else skip this
+    #   CUDA_LTOS=(?) # only if different, else skip this
     #   # strings `command -v nvcc` | \grep '\(sm\|compute\).*deprecate'
     #   CUDA_DEPRECATED=(?)
-    #
-    #   # https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html
-    #   # https://docs.nvidia.com/deploy/cuda-compatibility/index.html also shows the actual minimal versions
-    #
-    #   if [ "${OS-}" == "Windows_NT" ]; then
-    #     CUDA_MINIMAL_DRIVER_VERSION=?
-    #   else
-    #     CUDA_MINIMAL_DRIVER_VERSION=?
-    #   fi
-    #   ;;
 
     *)
       echo "CUDA Version ${CUDA_VERSION} is not programmed in here yet..." >&2
       CUDA_ARCHES=()
       CUDA_CODES=()
-      CUDA_MINIMAL_DRIVER_VERSION=0
       ;;
   esac
+
+  if [ "${#CUDA_CODES[@]}" = "0" ]; then
+    CUDA_CODES=("${CUDA_ARCHES[@]}")
+  fi
+
+  # Starting in 11.0, lto architectures were added
+  if [[ ${1} = 11.* ]] && [ "${#CUDA_LTOS[@]}" = "0" ]; then
+    CUDA_LTOS=("${CUDA_ARCHES[@]}")
+  fi
+
+  local versions
+
+  case "${1}" in
+    1.0*)
+      versions=(100.14 162.01) #~
+      ;;
+    1.1*)
+      versions=(169.01 169.01)
+      ;;
+    2.0*)
+      versions=(177.70 177.70) #~
+      ;;
+    2.1*)
+      versions=(180.22 180.22) #~
+      ;;
+    2.2*)
+      versions=(185.18.14 185.18.14) #~
+      ;;
+    2.3*)
+      versions=(190.53 190.53) #~
+      ;;
+    3.0*)
+      versions=(195.36.15 195.36.15)
+      ;;
+    3.1*)
+      versions=(256.40 256.40)
+      ;;
+    3.2*) # CUDA .16
+      versions=(260.19.26 260.19.26)
+      ;;
+    4.0*) # CUDA .17
+      versions=(270.41.19 270.41.19)
+      ;;
+    4.1*) # CUDA .28
+      versions=(285.05.33 285.05.33)
+      ;;
+    4.2*) # CUDA .9
+      versions=(295.41 295.41)
+      ;;
+    5.0*) # CUDA .35
+      versions=(304 304)
+      ;;
+    5.5*) # .0, CUDA .22
```

Comment:
what are these numbers?

Reply:
5.5.0 is CUDA 5.5.22. It's kind of confusing, so I make note of it.

Prior to 11, the CUDA version is retrieved when you run `nvcc --version`, so for "CUDA" version 5.5.0 (the first version of CUDA 5.5) the CUDA release version is 5.5.22.

However, around version 11, they no longer set the CUDA version in the individual components. Instead they are versioned independently.

Since this function is designed to "take the CUDA version" which can either be user selected _or_ auto detected, it has to handle all the different variants. So in this case `5.5`, `5.5.0`, and `5.5.22`

So far, this discrepancy has never created a conflict, but there is no guarantee that moving forward NVIDIA will never make a conflict between these two. For now, it works out.